tests/integration/test_auth.py

```diff
@@ -0,0 +1,67 @@
+from . import BaseIntegrationTest, default_users
+from datameta import models
+from datameta.api import base_url
+
+class TestApiKeyUsageSenario(BaseIntegrationTest):
+    """
+    Tests ApiKey creation, usage, and deletion.
+    """
+
+    def step_1_req_apikey(self):
+        """Request ApiKey"""
+        request_body = {
+            "email": self.state["user"].email,
+            "password": self.state["user"].password,
+            "label": self.state["apikey_label"]
+        }
+        
+        response = self.testapp.post_json(
+            base_url + "/keys", 
+            params=request_body, 
+            status=200
+        ) 
+        
+        self.state["apikey_response"] = response.json
+
+    def step_2_list_apikeys(self):
+        """Get a list of all ApiKeys"""
+        # request params:
+        user_id = self.state["user"].uuid
+        token = self.state["apikey_response"]["token"] # previously created apikey 
+        request_headers = {
+            "Authorization": f"Bearer {token}"
+        }
+
+        response = self.testapp.get(
+            base_url + f"/users/{user_id}/keys",
+            headers=request_headers,
+            status=200
+        ) 
+
+        # check whether current response is consistent with the previous responce
+        # obtained when creating the api keys
+        keys_to_compare = ["apiKeyId", "label", "expiresAt"]
```

Comment:
This will be renamed in the ID refactoring PR (`apiKeyId` → `id`) and will be of type `Identifier`, but changing that should be handled there then